File: leetcode/Solution242.java
public class Solution242 {
    public boolean isAnagram(String s, String t) {
        int[] count = new int[26];
        for(int i=0; i<s.length(); i++)
            count[s.charAt(i)-'a']++;
        for(int i=0; i<t.length(); i++){
            if(count[t.charAt(i)-'a']>0)
                count[t.charAt(i)-'a']--;
            else
                return false;
        }
        
        for(int i: count)
            if(i>0)
                return false;
        return true;
    }
}

File: leetcode/Solution242.py
def isAnagram(self, s, t):
        # populate the countArray
        count = [0] *26
        for i in s:
            count[ord(i)-ord("a")]+=1
            
        # remove from count according to t
        for i in t:
            if count[ord(i)-ord("a")]>0:
                count[ord(i)-ord("a")]-=1
            else:
                # no need to continue since we have an extra character in t
                return False 
        
        # check to make sure count = 0;
        if sum(count)==0:
            return True;
        return False